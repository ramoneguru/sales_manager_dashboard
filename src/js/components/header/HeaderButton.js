/**
 *
 * @author Drew Robinson (hello@drewrobinson.com)
 * @version 0.0.1
 * @param props
 * @exports HeaderButton Constructor
 */

import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

/**
 * Represents header button (hamburger)
 * @param props
 * @returns {XML}
 * @constructor
 * @desc prefer normal functions (not arrow functions) for stateless/refless components over over classes
 */
function HeaderButton(props){
  var styles = classNames('button', props.active === 1 ? 'active' : '');

  return (
    <div className={styles} onClick={props.handler}>
      <div className="bar bar1"></div>
      <div className="bar bar2"></div>
      <div className="bar bar3"></div>
    </div>
  )
}

HeaderButton.propTypes = {
  "active"  : PropTypes.number.isRequired,
  "handler" : PropTypes.func.isRequired
};

export default HeaderButton;