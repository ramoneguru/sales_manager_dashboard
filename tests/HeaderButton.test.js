import React from 'react';
import { shallow, render, mount } from 'enzyme';
import HeaderButton from '../src/js/components/header/HeaderButton';


describe('HeaderButton component ', function(){
  "use strict";

  it('should render correctly', () => {
    let menuClickHandler = function(){
      "use strict";
    };

    let active = 0;

    const wrapper = shallow(
      <HeaderButton active={ active } handler={ menuClickHandler }></HeaderButton>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should have active prop of type number', () => {
    let menuClickHandler = function(){
      "use strict";
    };

    let active = 0;

    const wrapper = mount(
      <HeaderButton active={ active } handler={ menuClickHandler }></HeaderButton>
    );

    expect(typeof wrapper.prop('active')).toEqual('number');
  });

  it('should have handler prop of type function', () => {
    let menuClickHandler = function(){
      "use strict";
    };

    let active = 0;

    const wrapper = mount(
      <HeaderButton active={ active } handler={ menuClickHandler }></HeaderButton>
    );

    expect(typeof wrapper.prop('handler')).toEqual('function');
  });


});
