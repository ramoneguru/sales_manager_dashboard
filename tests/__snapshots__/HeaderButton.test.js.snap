// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton component  should render correctly 1`] = `
<div
  className="button"
  onClick={[Function]}
>
  <div
    className="bar bar1"
  />
  <div
    className="bar bar2"
  />
  <div
    className="bar bar3"
  />
</div>
`;
